app.js

```diff
@@ -9,8 +10,30 @@ app.get("/api", (req, res, next) => {
     res.status(200).send(endpoints)
 })
 
+app.get("/api/articles/:article_id", getArticleById);
+
+
 app.all("*", (req, res, next) => {
     res.status(404).send({message: "path not found"});
 })
 
+
+app.use((error, req, res, next) => {
```

Comment:
Error handling middleware is looking great - you could extract these out to a separate file in line with mvc conventions as per the NC error handling notes if you wanted to tidy up your app!